### PR TITLE
Us30313233

### PR DESCRIPTION
**Pull request by Lithnotep**
#add  US 30 completion
#add  US 31 completion with adjusted specs
#add  US 32 completion with adjusted specs
#add  US 31 completion with adjusted specs and Admin views adjustment

### CODE REVIEW

File: app/views/admin/dashboard/index.html.erb [hunk 1]
```diff
@@ -1,14 +1,57 @@
...
+      <h2><%= "Order: #{order.id}" %></h2>
+      <p>User: <%= link_to "#{order.user.name}", "/admin/users/#{order.user.id}" %></p>
+      <p><%="Created on: #{order.created_at}"%></p>
+      <% if order.status == "packaged" %>
+        <%=button_to("Ship", "/orders/#{order.id}/ship", method: "patch")%>
+      <% end %>
+    <% end %>
+    </section>
+  <% end %>
+
```
Comment (jpc20) [inline app/views/admin/dashboard/index.html.erb:56]: Nice. Let be sure to check this out on heroku.

File: spec/features/orders/show_spec.rb [hunk 2]
```diff
@@ -36,13 +37,34 @@
...
-      expect(Order.last.status).to eq("cancelled")
-      expect(ItemOrder.first.status).to eq("unfulfilled")
-      expect(ItemOrder.last.status).to eq("unfulfilled")
+      @order_1.reload
+      @tire.reload
+      @pull_toy.reload
+      expect(@order_1.status).to eq("cancelled")
+      expect(@order_1.item_orders.first.status).to eq("unfulfilled")
+      expect(@order_1.item_orders.last.status).to eq("unfulfilled")
+      expect(@tire.inventory).to eq(14)
```
Comment (janegreene) [inline spec/features/orders/show_spec.rb:60]: Nice thorough testing.

File: app/controllers/merchant/item_orders_controller.rb [hunk 3]
```diff
@@ -2,9 +2,11 @@ class Merchant::ItemOrdersController < Merchant::BaseController
   def update
     item_order = ItemOrder.find(params[:id])
     item = Item.find(params[:item_id])
+    order = Order.find(item_order.order_id)
       item.update_attribute(:inventory, item.inventory - item_order.quantity)
       item_order.update_attribute(:status, "fulfilled")
       flash[:success] = "You have fulfilled #{item.name}"
+      order.packaged_check
```
Comment (janegreene) [inline app/controllers/merchant/item_orders_controller.rb:9]: This method lives in order.rb and runs anytime an item is added to an order to ensure new items get packaged.

**Review by janegreene — APPROVED** [review]
Completion of stories 30-33

**Pull request merged**